picard/ui/profileeditor.py

```diff
@@ -0,0 +1,455 @@
+        information to the user settings, and close the profile editor dialog.
+        """
+        # Save any unsaved changes
+        if self.ui.profile_list.count():
+            self.save_profile(silent=True)
+
+        all_profiles = list(self._all_profiles())
+        all_profile_ids = set(x['id'] for x in all_profiles)
+        keys = set(deepcopy(self.profile_settings).keys())
+        for id in keys:
+            if id not in all_profile_ids:
+                del self.profile_settings[id]
+
+        config = get_config()
+        config.profiles[self.PROFILES_KEY] = all_profiles
+        config.profiles[self.SETTINGS_KEY] = self.profile_settings
+
+        self.main_window.enable_renaming_action.setChecked(config.setting["rename_files"])
+        self.main_window.enable_moving_action.setChecked(config.setting["move_files"])
+        self.main_window.enable_tag_saving_action.setChecked(not config.setting["dont_write_tags"])
+
+        self.close()
+
+    def closeEvent(self, event):
+        """Custom close event handler to save editor settings.
+        """
+        config = get_config()
+        config.persist[self.POSITION_KEY] = self.ui.profile_list.currentRow()
+        config.persist[self.EXPANDED_KEY] = self.expanded_sections
+        super().closeEvent(event)
+
+    def _all_profiles(self):
+        """Get all profiles from the profiles list in order from top to bottom.
+
+        Yields:
+            dict: Profile information in a format for saving to the user settings
+        """
+        for row in range(self.ui.profile_list.count()):
+            item = self.ui.profile_list.item(row)
+            yield item.get_dict()
+
+    def set_button_states(self):
+        """Set the enabled / disabled states of the buttons.
+        """
+        state = self.current_profile_id is not None
+        self.save_profile_button.setEnabled(state)
+        self.copy_profile_button.setEnabled(state)
+        self.delete_profile_button.setEnabled(state)
+
+    def get_checked_items_from_tree(self):
+        """Get the keys for the settings that are checked in the profile settings tree.
+
+        Yields:
+            str: Settings key
+        """
+        for i in range(self.ui.settings_tree.topLevelItemCount()):
+            tl_item = self.ui.settings_tree.topLevelItem(i)
+            for j in range(tl_item.childCount()):
+                item = tl_item.child(j)
+                if item.checkState(0) == QtCore.Qt.Checked:
+                    yield item.data(0, QtCore.Qt.UserRole)
+
+    def unsaved_changes_confirmation(self):
+        """Check if there are unsaved changes and ask the user to confirm the action resulting in their loss.
+
+        Returns:
+            bool: True if no unsaved changes or user confirms the action, otherwise False.
+        """
+        if not self.loading and self.current_profile_id and self.profile_settings_changed() and not self.confirmation_dialog(
```

Comment:
In `current_item_changed()` there's already a test against `self.loading`, imho it makes more sense to test for it before calling this method anyway. Unsure.

Reply:
You're right.  It's no longer necessary here.